app/src/main/java/ch/epfl/culturequest/ProfileCreatorActivity.java

```diff
@@ -106,34 +109,44 @@ public void createProfile(View view) {
         profile.setUsername(username);
         profile.setUid(Authenticator.getCurrentUser().getUid());
 
-        //if user is anonymous, we don't want to store the profile in the database
-        if (!Authenticator.getCurrentUser().isAnonymous()) {
-
-            //if the profile picture is not the default one, we store it in the storage
-            if (!profilePicUri.equals(ProfileUtils.DEFAULT_PROFILE_PIC_PATH))
-                FireStorage.uploadNewProfilePictureToStorage(profile, profilePicBitmap, true).whenComplete(
-                        (profile, throwable) -> {
-                            if (throwable != null) {
-                                throwable.printStackTrace();
-                            } else {
-                                Database.setProfile(profile);
-                                Profile.setActiveProfile(profile);
+        // Get first the device token, then store the profile in the database if it is not anonymous
+        FireMessaging.getDeviceToken().whenComplete((token, ex) -> {
+            if (ex == null) {
+                List<String> deviceTokens = new ArrayList<>();
+                deviceTokens.add(token);
+                profile.setDeviceTokens(deviceTokens);
```

Comment:
Same comment as in `Authenticator.java`

Reply:
I don't get what is the problem here ?

Reply:
What I was saying is that here you just check the throwable but you don't check if the token is already in the list, is that because the profile is being created?

Reply:
yes it is the first time the users connects everything is empty ;)